build-tools/docker/build.py

```diff
@@ -0,0 +1,99 @@
+import toml
+import os
+import subprocess
+import argparse
+
+
+def get_cargo_version(cargo_toml_path):
+    if not os.path.exists(cargo_toml_path):
+        raise ValueError(f"No such file: {cargo_toml_path}")
+
+    # Read the Cargo.toml file
+    config = toml.load(cargo_toml_path)
+
+    # Get the version
+    version = config.get('package', {}).get('version')
+
+    if version is None:
+        raise ValueError(f"No version specified in {cargo_toml_path}")
+
+    return version
+
+def build_docker_image(dockerfile_path, image_name, version):
+    # Docker build command
+    command = f"docker build -t {image_name}:{version} -f {dockerfile_path} ."
+
+    # Run the command
+    process = subprocess.Popen(command, shell=True)
+    output, error = process.communicate()
+
+    if error:
+        print(f"Error occurred: {error}")
+    else:
+        print(f"Built {image_name}:{version} successfully.")
```

Comment:
Shouldn't we stop the script if `docker build` fails? (i.e. why not just use `check_call` here like in other places?)